yarn-project/simulator/src/avm/journal/trace.ts

```diff
@@ -1,176 +1,278 @@
+import { AvmContractInstanceHint, AvmExecutionHints, AvmExternalCallHint, AvmKeyValueHint, AztecAddress, ContractStorageRead, ContractStorageUpdateRequest, EthAddress, Gas, L2ToL1Message, LogHash, NoteHash, Nullifier, ReadRequest } from '@aztec/circuits.js';
+import { UnencryptedFunctionL2Logs, UnencryptedL2Log } from '@aztec/circuit-types';
+import { ContractInstanceWithAddress } from '@aztec/types/contracts';
+import { EventSelector } from '@aztec/foundation/abi';
 import { Fr } from '@aztec/foundation/fields';
+import { type FieldsOf } from '@aztec/foundation/types';
 
+import {
+  type PublicExecution,
+  type PublicExecutionResult,
+} from '../../public/execution.js';
 import {
   type TracedContractInstance,
-  type TracedL1toL2MessageCheck,
-  type TracedNoteHash,
-  type TracedNoteHashCheck,
-  type TracedNullifier,
-  type TracedNullifierCheck,
-  type TracedPublicStorageWrite,
-  type TracedUnencryptedL2Log,
 } from './trace_types.js';
-import { AvmExecutionHints, AvmKeyValueHint, AztecAddress, ContractStorageRead } from '@aztec/circuits.js';
+import { AvmRevertReason } from '../errors.js';
+import { createSimulationError } from '../../common/errors.js';
+
+export type TraceNestedCallArgs = [
+  /** The execution that triggered this result. */
+  /*execution:*/PublicExecution,
```

Comment:
It feels strange to have the simulator, and the journal, to be aware of the PublicExecution object/concept...